src/components/LHNOptionsList/OptionRowLHN.js

```diff
@@ -52,14 +60,25 @@ const defaultProps = {
     onSelectRow: () => {},
     isFocused: false,
     style: null,
+    comment: '',
 };
 
 function OptionRowLHN(props) {
     const optionItem = SidebarUtils.getOptionData(props.reportID);
 
-    if (!optionItem) {
+    const runUseEffectOnlyOnce = useCallback(() => {
+        if (!optionItem.hasDraftComment && props.comment && props.comment.length > 0 && !props.isFocused) {
+            Report.setReportWithDraft(props.reportID, true);
+        }
         return null;
-    }
+    }, [optionItem, props.comment, props.isFocused, props.reportID]);
+
+    useEffect(() => {
+        if (!optionItem) {
+            return null;
+        }
+        runUseEffectOnlyOnce();
+    }, [runUseEffectOnlyOnce, optionItem]);
```

Comment:
If `runUseEffectOnlyOnce` is only run once, why don't we place the function inside the `useEffect`?

E.g:
```js
    useEffect(() => {
        const updateReportDraft = () => {
            if (!optionItem || optionItem.hasDraftComment || !props.comment || props.comment.length <= 0 || props.isFocused) {
                return;
            }


            Report.setReportWithDraft(props.reportID, true);
        }

        updateReportDraft();
    }, []);

    if (!optionItem) {
        return null;
    }
```